### PR TITLE
Support reply as a message reference for !raw [json]

### PR DESCRIPTION
**Pull request by ichard26**
Resolves GH-2761.

image

### CODE REVIEW

**Comment by ichard26** [comment]
I didn't break the referenced message fallback out into its own function as I am not convinced it would be a net readability improvement. I don't have a strong opinion on this though.

File: bot/exts/info/information.py [hunk 1]
```diff
@@ -524,13 +525,29 @@ def add_content(title: str, content: str) -> None:
...
+    async def raw(self, ctx: Context, message: Message | None = None) -> None:
         """Shows information about the raw API response."""
+        if message is None:
+            if (reference := ctx.message.reference) and isinstance(reference.resolved, Message):
+                message = reference.resolved
+
+        if message is None:
+            await send_denial(ctx, "Missing message argument. Please provide a message ID/link or reply to a message.")
+            return
+
```
Comment (MrHemlock) [inline bot/exts/info/information.py:537]: Is there a particular reason we're doing this in two checks?
```suggestion
        if message is None:
            if (reference := ctx.message.reference) and isinstance(reference.resolved, Message):
                message = reference.resolved

            else:
                await send_denial(ctx, "Missing message argument. Please provide a message ID/link or reply to a message.")
                return

```

File: bot/exts/info/information.py [hunk 2]
```diff
@@ -524,13 +525,29 @@ def add_content(title: str, content: str) -> None:
...
+    async def json(self, ctx: Context, message: Message | None = None) -> None:
         """Shows information about the raw API response in a copy-pasteable Python format."""
+        if message is None:
+            if (reference := ctx.message.reference) and isinstance(reference.resolved, Message):
+                message = reference.resolved
+
+        if message is None:
+            await send_denial(ctx, "Missing message argument. Please provide a message ID/link or reply to a message.")
+            return
+
```
Comment (MrHemlock) [inline bot/exts/info/information.py:550]: Same as above, is there a specific reason for making these separate checks?


```suggestion
        if message is None:
            if (reference := ctx.message.reference) and isinstance(reference.resolved, Message):
                message = reference.resolved

            else:
                await send_denial(ctx, "Missing message argument. Please provide a message ID/link or reply to a message.")
                return

```

**Review by MrHemlock — CHANGES_REQUESTED** [review]
Not sure if I'm being specifically picky or not, but it just seems redundant to check `if message is None:` twice

**Review by wookie184 — APPROVED** [review]
All looks good to me

**Comment by ChrisLovering** [comment]
we should really get around to writing this as a d.py convertor in bot-core one of these days. This pattern seems to be used quite often now.

**Pull request merged**